### DISCUSSION TITLE
Managing permissions for certain kernel profiles

### DISCUSSION THREAD

**Discussion by OrenZ1** [jupyter/enterprise_gateway#1386] [Q&A] [answered]
Hi, I was wondering if there is a way to prevent / allow certain users to create certain kernels which are present on an EG? 
i.e. if I have an EG running with 2 kernelspecs available - 1 for python and one for R, but I want to prevent a certain user which connects to this EG too from creating a python kernel.

I am aware of the “authorized_users” and “unauthorized_users” config options, but they are only on the whole EG scope, and I didn’t find a way to utilize these options for my use case above.
Thanks in advance!

**Comment by kevin-bates** [comment, marked as answer]
Hi @OrenZ1,  See the section [Per-kernel Overrides](https://jupyter-enterprise-gateway.readthedocs.io/en/latest/operators/config-kernel-override.html#per-kernel-overrides) in the Operators guide of the docs.  User authorization is one of config settings that can be overridden within the kernelspec.  There's an example of this in the [Process-proxy Configuration](https://jupyter-enterprise-gateway.readthedocs.io/en/latest/contributors/system-architecture.html#process-proxy-configuration) section of the Contributors guide as well.

Keep in mind when using `unauthorized_users` lists that the global settings will always take precedence for denials.

**Reply by OrenZ1** [reply]
Thank you, for some reason I didn’t see that! It works perfectly.
